internal/storage/memory/memory.go

```diff
@@ -137,9 +137,9 @@ func (s *Storage) gc() {
 	}
 }
 
-// Return database client
-func (s *Storage) Conn() map[string]entry {
-	s.mux.RLock()
-	defer s.mux.RUnlock()
-	return s.db
+// ConnLocked can be used to access the underlying db in a thread-safe manner.
+func (s *Storage) ConnLocked(f func(map[string]entry)) {
```

Comment:
This is an interface, we shouldn't rename it to avoid breaking changes.

Reply:
@gaby What interface does this implement?

I can see the earlier `Conn` method only on this struct
https://sourcegraph.com/search?q=context:global+repo:https://github.com/gofiber/fiber/+%22Conn()%22&patternType=standard&sm=1&groupBy=repo

Reply:
https://github.com/gofiber/storage/blob/main/storage.go

We may need to discuss this, since it's implemented differently in multiple places 🤔

Reply:
As far as I understand, this method is only provided by `memory.Storage`, because I don't think any other storage will return a `Conn` of type  `map[string]entry`.

Reply:
no this method is provided by all adapters
https://github.com/search?q=repo%3Agofiber%2Fstorage%20conn&type=code

Reply:
I see

I think we can make `Storage.db` a `sync.Map` so we can return it as it is from `Conn()`, without worrying about thread safety.

Thoughts @gaby , @ReneWerner87 ?

Reply:
A ping on this in case it was missed. @gaby @ReneWerner87

Reply:
sync.Map is ok, but slower or ?

Reply:
I can run the `Benchmark_Storage_Memory` and see if it's better or worse.